### ISSUE TITLE
MaterializeMySQL     DB::Exception: Sorting key cannot contain nullable columns

### ISSUE THREAD

**Issue by chang0391** [ClickHouse/ClickHouse#14605]
**Describe the bug**
when I execute create table in mysql, I did not find the table in CH.
when I execute select for other table in CH,I got the issue 
```
test-1-118.raipeng.com :) select *from aa
SELECT *
FROM aa
Received exception from server (version 20.8.1):
Code: 44. DB::Exception: Received from localhost:9000. DB::Exception: Sorting key cannot contain nullable columns. 
```
**How to reproduce**
step1:
```
mysql>create table test(id bigint auto_increment,name varchar(11),primary key(id))
```
step2:
```
test-1-118.raipeng.com :) select *from test
SELECT *
FROM test
Received exception from server (version 20.8.1):
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table just.test doesn't exist.. 
test-1-118.raipeng.com :) select *from aa
SELECT *
FROM aa
Received exception from server (version 20.8.1):
Code: 44. DB::Exception: Received from localhost:9000. DB::Exception: Sorting key cannot contain nullable columns. 
```
error log as below:
```
 (version 20.8.1.4513 (official build))
2020.09.09 10:53:56.506421 [ 61597 ] {2236a5c2-e26b-4d9f-84d0-261d0a013e1a} <Error> executeQuery: Code: 44, e.displayText() = DB::Exception: Sorting key cannot contain nullable columns (version 20.8.1.4513 (official build)) (from [::1]:35198) (in query: select *from aa), Stack trace (when copying this message, always include the lines below):

0. /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1a807d30 in /usr/lib/debug/usr/bin/clickhouse
1. /build/obj-x86_64-linux-gnu/../src/Common/Exception.cpp:37: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0xff9e75d in /usr/lib/debug/usr/bin/clickhouse
2. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134: DB::checkKeyExpression(DB::ExpressionActions const&, DB::Block const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool) (.cold) @ 0x17bf2166 in /usr/lib/debug/usr/bin/clickhouse
3. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134: DB::MergeTreeData::checkProperties(DB::StorageInMemoryMetadata const&, DB::StorageInMemoryMetadata const&, bool) const @ 0x17bd1de6 in /usr/lib/debug/usr/bin/clickhouse
4. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/memory:3028: DB::MergeTreeData::MergeTreeData(DB::StorageID const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::StorageInMemoryMetadata const&, DB::Context&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::MergeTreeData::MergingParams const&, std::__1::unique_ptr<DB::MergeTreeSettings, std::__1::default_delete<DB::MergeTreeSettings> >, bool, bool, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>) @ 0x17beeaa4 in /usr/lib/debug/usr/bin/clickhouse
5. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/memory:2615: DB::StorageMergeTree::StorageMergeTree(DB::StorageID const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::StorageInMemoryMetadata const&, bool, DB::Context&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::MergeTreeData::MergingParams const&, std::__1::unique_ptr<DB::MergeTreeSettings, std::__1::default_delete<DB::MergeTreeSettings> >, bool) @ 0x179e218f in /usr/lib/debug/usr/bin/clickhouse
6. /build/obj-x86_64-linux-gnu/../base/common/../ext/shared_ptr_helper.h:19: DB::create(DB::StorageFactory::Arguments const&) @ 0x17de7abc in /usr/lib/debug/usr/bin/clickhouse
7. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/functional:1715: std::__1::__function::__func<std::__1::shared_ptr<DB::IStorage> (*)(DB::StorageFactory::Arguments const&), std::__1::allocator<std::__1::shared_ptr<DB::IStorage> (*)(DB::StorageFactory::Arguments const&)>, std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()(DB::StorageFactory::Arguments const&) @ 0x17deb8e7 in /usr/lib/debug/usr/bin/clickhouse
8. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134: DB::StorageFactory::get(DB::ASTCreateQuery const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, DB::Context&, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, bool) const @ 0x1795b601 in /usr/lib/debug/usr/bin/clickhouse
9. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/type_traits:3696: DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&) @ 0x17380536 in /usr/lib/debug/usr/bin/clickhouse
10. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterCreateQuery.cpp:690: DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) @ 0x17382cbd in /usr/lib/debug/usr/bin/clickhouse
11. /build/obj-x86_64-linux-gnu/../src/Interpreters/InterpreterCreateQuery.cpp:874: DB::InterpreterCreateQuery::execute() @ 0x17385809 in /usr/lib/debug/usr/bin/clickhouse
12. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:389: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x177d4432 in /usr/lib/debug/usr/bin/clickhouse
13. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:675: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0x177d5d5f in /usr/lib/debug/usr/bin/clickhouse
14. /build/obj-x86_64-linux-gnu/../src/Interpreters/MySQL/InterpretersMySQLDDLQuery.h:71: DB::InterpreterExternalDDLQuery::execute() @ 0x17440f1d in /usr/lib/debug/usr/bin/clickhouse
15. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:389: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x177d4432 in /usr/lib/debug/usr/bin/clickhouse
16. /build/obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:675: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0x177d5d5f in /usr/lib/debug/usr/bin/clickhouse
17. /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134: DB::tryToExecuteQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x17397bd4 in /usr/lib/debug/usr/bin/clickhouse
18. /build/obj-x86_64-linux-gnu/../src/Databases/MySQL/MaterializeMySQLSyncThread.cpp:618: DB::MaterializeMySQLSyncThread::onEvent(DB::MaterializeMySQLSyncThread::Buffers&, std::__1::shared_ptr<DB::MySQLReplication::EventBase> const&, DB::MaterializeMetadata&) @ 0x1739ebb6 in /usr/lib/debug/usr/bin/clickhouse
```

**Comment by filimonov** [comment]
Duplicate of #5319

**Comment by zhang2014** [comment]
The error about MaterializedMySQL engine has been fixed in #14397

**Comment by zhang2014** [comment]
Maybe we close this issue?
CC: @filimonov @tavplubix
